staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -0,0 +1,998 @@
+	CountModeExact = "Exact"
+	CountModeAll   = "All"
+)
+
+// Exactly one field must be set.
+type Selector struct {
+	// CEL contains a CEL expression for selecting a device.
+	//
+	// +required
+	CEL *CELSelector `json:"cel,omitempty" protobuf:"bytes,1,opt,name=cel"`
+}
+
+// CELSelector contains a CEL expression for selecting a device.
+type CELSelector struct {
+	// This CEL expression must evaluate to true if a device is suitable.
+	// This covers qualitative aspects of device selection.
+	//
+	// The language is as defined in
+	// https://kubernetes.io/docs/reference/using-api/cel/
+	// with several additions that are specific to device selectors.
+	//
+	// Attributes of a device are made available through a nested
+	// `device.attributes` map with the domain part of the attribute name
+	// as key in the outer map and the identifier as key in the inner
+	// map. All identifiers can be used in a field lookup:
+	//
+	//    device.attributes["dra.example.com"].driverVersion
+	//
+	// The type of each entry varies, depending on the attribute
+	// that is being looked up. The domain lookup returns an empty
+	// map if there is no attribute with that domain. However,
+	// unknown identifiers then trigger a runtime error.
+	//
+	// The `cel.bind` function is enabled and can be used to simplify
+	// expressions that access multiple attributes with the same domain:
+	//
+	//    cel.bind(dra, device.attributes["dra.example.com"], dra.someBool && dra.anotherBool)
+	//
+	// Capacities associated with a device are made available through a
+	// nested `device.capacities` map the same way as attributes.
+	//
+	// The `device.driverName` string variable can be used to check for a specific
+	// driver explicitly in a filter that is meant to work for devices from
+	// different vendors. It is provided by Kubernetes and matches the
+	// `driverName` from the ResourceSlice which provides the device.
+	//
+	// The CEL expression is applied to *all* available devices from any driver.
+	// The expression has to check for existence of an attribute when it is not
+	// certain that it is provided because runtime errors are not automatically
+	// treated as "don't select device". Instead, device selection fails completely
+	// and reports the error.
+	//
+	// Some more examples:
+	//
+	//    "memory" in device.capacities["dra.example.com"] && # Is the capacity available?
+	//       device.capacities["dra.example.com"].memory.isGreaterThan(quantity("1Gi")) # >= 1Gi
+	//
+	//    device.attributes["dra.example.com"].driverVersion.isGreaterThan(semver("1.0.0")) # >= v1.0.0, runtime error if not available
+	//
+	//    device.driverName == "dra.example.com" # any device from that driver
+	//
+	// +required
+	Expression string `json:"expression" protobuf:"bytes,1,name=expression"`
+}
+
+// Besides the request name slice, constraint must have exactly one field set.
+type Constraint struct {
+	// The constraint applies to devices in these requests. A single entry is okay
```

Comment:
This sentence is confusing to me,.  I will try to reword and see if you agree

"""
RequestNames is a list of the one or more requests in this claim which must co-satisfy this constraint.  If a request is fulfilled by multiple devices, then all of the devices must satisfy the constraint.  If this is not specified, this constraint applies to all requests in this claim.
"""

Reply:
That sounds right. Will us it.

Reply:
Done, and also renamed to just `Requests` - but see https://github.com/kubernetes/kubernetes/pull/125488#discussion_r1661220099.